python_src/tests/performance_manager/test_performance_manager.py

```diff
@@ -166,6 +167,60 @@ def mock__get_pyarrow_table(
     yield
 
 
+def flat_table_check(db_manager: DatabaseManager, service_date: int) -> None:
+    """checks to run on a flat table to ensure it has what would be expected"""
+    flat_table = generate_daily_table(db_manager, service_date)
+
+    # check that the shape is good
+    rows, columns = flat_table.shape
+    assert columns == 30
+
+    expected_row_count = db_manager.select_as_list(
+        sa.select(sa.func.count())
+        .where(
+            sa.and_(
+                VehicleEvents.service_date == service_date,
+                sa.or_(
+                    VehicleEvents.vp_move_timestamp.is_not(None),
+                    VehicleEvents.vp_stop_timestamp.is_not(None)
+                )
+            )
+        )
+    )[0]['count_1']
+    assert rows == expected_row_count
```

Comment:
this is the same where clause in the generated daily table function, but it checks that our joins don't add or remove anything.